features/admin.push-providers.v1/components/push-provider-card.tsx

```diff
@@ -0,0 +1,159 @@
+
+    const { t } = useTranslation();
+
+    const pushProviderFeatureFlagsConfig: FeatureFlagsInterface[] = useSelector(
+        (state: AppState) => state.config.ui.features.pushProviders.featureFlags
+    );
+
+    const featureFlag: string = useFeatureFlag(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP.APPLICATION_TEMPLATES,
+        pushProviderFeatureFlagsConfig
+    );
+
+    const featureStatus: PushProviderTemplateFeatureStatus = useMemo(() => {
+        if (!template?.customAttributes
+            || !Array.isArray(template?.customAttributes)
+            || template?.customAttributes.length === 0) {
+            return false;
+        }
+
+        const property: CustomAttributeInterface = template?.customAttributes?.find(
+            (property: CustomAttributeInterface) =>
+                property?.key === PushProviderConstants.FEATURE_STATUS_ATTRIBUTE_KEY
+        );
+
+        if (featureFlag === "TRUE") {
+            return property?.value as PushProviderTemplateFeatureStatus;
+        }
+
+        return property?.value;
+    }, [ template ]);
+
+    const resolveRibbonLabel = (featureStatus: PushProviderTemplateFeatureStatus): string => {
+        switch (featureStatus) {
+            case PushProviderTemplateFeatureStatus.NEW:
+                return t("common:new");
+        }
+    };
+
+    const resolveFeatureLabelClass = (featureStatus: PushProviderTemplateFeatureStatus) => {
+        switch (featureStatus) {
+            case PushProviderTemplateFeatureStatus.NEW:
+                return "oxygen-chip-new";
+        }
+    };
+
+    return(
+        <Card
+            key={ template?.id }
+            // className="push-provider-card"
+            className={ classnames("push-provider-card", {
+                "push-provider-card--selected": selected
+            }) }
+            onClick={
+                (e: MouseEvent<HTMLDivElement>) => {
+                    onClick(e);
+                }
+            }
+            data-componentid={ `${componentId}-${template?.id}` }
+        >
+            <CardContent className="push-provider-card-content">
+                <div className="push-provider-template-image-container">
+                    <img
+                        className="push-provider-template-image"
+                        src={
+                            ExtensionTemplateManagementUtils.resolveExtensionTemplateResourcePath(
+                                template?.image,
+                                ResourceTypes.NOTIFICATION_PROVIDERS
+                            )
+                        }
+                        loading="lazy"
+                    />
+                </div>
+                <div className="push-provider-template-name-label">
+                    <Typography variant="h6">
+                        { template?.name }
+                    </Typography>
+                </div>
+                {
+                    featureStatus
+                        ? (
```

Comment:
We can use the new FeatureFlag component as well https://github.com/wso2/identity-apps/blob/master/features/admin.feature-gate.v1/components/feature-flag-label.tsx

Reply:
In application template card also, we have used the old implementation[1]. Therefore did not replace the same implementation with new component for consistency.

[1] https://github.com/ZiyamSanthosh/identity-apps/blob/master-push-jsp/features/admin.application-templates.v1/components/application-template-card.tsx#L193